SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialEventTracingTest.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKTests/Public/GDKAutomationTestBase.h"
+#include "Tests/TestDefinitions.h"
+
+#include "Interop/Connection/SpatialTraceEventDataBuilder.h"
+
+#include "CoreMinimal.h"
+#include "Engine/Engine.h"
+
+#include <string>
+
+#define EVENTTRACING_TEST(TestName) GDK_AUTOMATION_TEST(Core, EventTracingTests, TestName)
+
+using namespace SpatialGDK;
+
+namespace
+{
+} // anonymous namespace
+
+EVENTTRACING_TEST(GIVEN_a_string_cache_WHEN_combing_two_string_THEN_the_strings_are_combined_and_stored_correctly)
+{
+	FSpatialTraceEventDataBuilder::FStringCache StringCache;
+	int32 Handle = StringCache.CombineStrings("string1", "string2");
+	bool bSuccess = FCStringAnsi::Strcmp("string1string2", StringCache.Get(Handle)) == 0;
+	TestTrue("String succesfully stored and retreived", bSuccess);
+	return true;
+}
+
+EVENTTRACING_TEST(
+	GIVEN_a_string_cache_WHEN_combing_two_string_that_will_overflow_but_cant_combine_THEN_the_strings_are_combined_and_stored_correctly)
+{
+	FSpatialTraceEventDataBuilder::FStringCache StringCache;
+
+	FString InputString;
+	for (int32 i = 0; i < StringCache.GetBufferSize() - 1; ++i)
+	{
+		InputString += "a";
+	}
+
+	std::string InputSrc = (const char*)TCHAR_TO_ANSI(*InputString);
+	int32 Handle = StringCache.CombineStrings(InputSrc.c_str(), "string2");
+	bool bSuccess = FCStringAnsi::Strcmp(InputSrc.c_str(), StringCache.Get(Handle)) == 0;
+
+	TestTrue("String succesfully stored and retreived", bSuccess);
+	return true;
+}
+
+EVENTTRACING_TEST(
+	GIVEN_a_string_cache_WHEN_combing_two_string_that_will_overflow_but_can_partially_combine_THEN_the_strings_are_combined_and_stored_correctly)
+{
+	FSpatialTraceEventDataBuilder::FStringCache StringCache;
+
+	FString InputString;
+	for (int32 i = 0; i < StringCache.GetBufferSize() - 2; ++i)
+	{
+		InputString += "a";
+	}
+
+	FString OutputString = InputString + "s";
+	std::string InputSrc = (const char*)TCHAR_TO_ANSI(*InputString);
```

Comment:
As above no need to store this in a std::string, FString should already do the lifetime management you need in this case.